ci-operator/step-registry/openstack/deprovision/deletednsrecords/openstack-deprovision-deletednsrecords-commands.sh

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env bash
+
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+
+
+set -e
+export AWS_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/.awscred"
+export AWS_DEFAULT_REGION=us-east-1
+export AWS_DEFAULT_OUTPUT=json
+export AWS_PROFILE=openshift-ci-infra
+
+
+
+HOSTED_ZONE_ID=$(aws route53 list-hosted-zones-by-name --dns-name "${BASE_DOMAIN}" | python -c '
+import json,sys;
+print (json.load(sys.stdin)["HostedZones"][0]["Id"].split("/")[-1])'
+)
+
+for RECORD_FILE in api-record.json ingress-record.json
+do
+    if [[ -f "${SHARED_DIR}/${RECORD_FILE}" ]]; then
+        echo Deleting ${RECORD_FILE}
+        sed -e's/UPSERT/DELETE/g' ${SHARED_DIR}/${RECORD_FILE} > ${SHARED_DIR}/delete-${RECORD_FILE}
+        sed -i -e's/Create/Delete/g' ${SHARED_DIR}/delete-${RECORD_FILE}
```

Comment:
sed's in-place editing is implemented inconsistently across platforms; to stay on the safe side, you could use the single-pass version of this command we have in the UPI job

https://github.com/openshift/release/blob/ada8e843bc2b19260856e784b685a4975e790bdb/ci-operator/templates/openshift/installer/cluster-launch-installer-openstack-upi-e2e.yaml#L1215-L1218

Reply:
It should be a very specific platform that executes this, but better safe than sorry